pilot/pkg/networking/core/v1alpha3/cluster_tls.go

```diff
@@ -168,14 +168,14 @@ func (cb *ClusterBuilder) buildUpstreamClusterTLSContext(opts *buildClusterOpts,
 		// The code has repeated snippets because We want to use predefined alpn strings for efficiency.
 		if cb.isHttp2Cluster(c) {
 			// This is HTTP/2 in-mesh cluster, advertise it with ALPN.
-			if features.MetadataExchange {
+			if features.MetadataExchange && !features.DisableMxALPN {
```

Comment:
Q: my understanding was that we can send the metadata exchange even for plain text - and with workloads that don't use istio ALPN ( example proxyless gRPC, etc). But it looks I may be wrong ? Is there some other place where MX is enabled ?

Reply:
Yes, it should work for plaintext. We may not have the right test to exercise this though - with all the different ways to get telemetry, it's hard to validate the discovery works right.